site/docs/v3-adoption-progress.md

```diff
@@ -0,0 +1,173 @@
+| **Spark** | ❌ **Not Implemented** | ❌ **Not Implemented** | Only supports microsecond precision timestamps |
+| **Flink** | ✅ **Full Support** | ✅ **Full Support** | Complete support for all timestamp types and precisions |
+
+### Unknown Type Support
+
+| **Engine** | **Core Support** | **Implementation Status** | **Notes** |
+|---|---|---|---|
+| **Core** | ✅ **Full Support** | Complete type system and schema evolution support | Supports optional unknown fields, proper validation |
+| **Spark v4.0+** | ✅ **Full Support** | Complete type mapping to `NullType` | Full support with comprehensive tests |
+| **Spark v3.4/v3.5** | ❌ **Not Implemented** | Missing from type conversion | Throws "Cannot convert unknown type to Spark" error |
+| **Flink** | ✅ **Full Support** | Complete type mapping to `NullType` | Full support across all versions (v1.20, v2.0, v2.1) |
+
+### Multi-Argument Transforms Support
+
+| **Engine** | **Z-Order Support** | **Implementation Status** | **Notes** |
+|---|---|---|---|
+| **Core** | ✅ **Full Support** | Complete `Zorder` class implementation | Full support for multi-column transforms in partitioning and sorting |
+| **Spark** | ✅ **Full Support** | Complete `SparkZOrderFileRewriteRunner` implementation | Full support across all versions (v3.4, v3.5, v4.0) with file rewriting |
+| **Flink** | ❌ **Not Implemented** | No multi-argument transform support | Only single-argument transforms supported |
+
+### Default Values Support
+
+| **Engine** | **initial-default** | **write-default** | **Notes** |
+|---|---|---|---|
+| **Core** | ✅ **Full Support** | ✅ **Full Support** | Complete support for both types with schema evolution |
+| **Spark** | ✅ **Full Support** | ✅ **Full Support** | Full support across all versions (v3.4, v3.5, v4.0) |
+| **Flink** | ✅ **Full Support** | ✅ **Full Support** | Full support across all versions (v1.20, v2.0, v2.1) |
+
+## Critical Missing Features
+
+### 1. Geometry and Geography Types
+- **Status**: ❌ **Not Implemented** across all engines
+- **Issue**: While type definitions exist in the API module, no engine has actual data processing implementation
+- **Impact**: Spatial data operations are not supported
+
+### 2. Table Encryption
+- **Status**: ❌ **Only Core supports it**
+- **Issue**: Spark and Flink don't support AES-GCM table encryption
+- **Impact**: Security features are limited to Core usage
+
+### 3. Variant Type
+- **Status**: ⚠️ **Partially Implemented in Core**
+- **Issue**: Core supports Avro and Parquet but not ORC; Spark and Flink have no support
+- **Impact**: Semi-structured data handling is limited to specific file formats
+
+### 4. Nanosecond Timestamps
+- **Status**: ❌ **Spark doesn't support them**
+- **Issue**: Spark connector doesn't support nanosecond precision timestamps
+- **Impact**: High-precision timestamp operations are not available in Spark
+
+## Recommendations for Users
```

Comment:
I still feel like we should let users just look at the "V3 feature implementation status" table above instead of the project giving recommendations like this. It's also another part of the doc that we'll need to update as support gets added (e.g. there's an open PR out for Flink DV write support right now), so there's going to be a fair amount of churn in this section.

We can avoid that by just having that single table above.

Reply:
Sounds good!